### PR TITLE
Interval class for a set of continuous (real) numbers

### PR DESCRIPTION
**Pull request by fdodino**
See issue #738 
Also fixed a copy/paste problem in WollokJavaConversions for double.

### CODE REVIEW

**Comment by tesonep** [comment]
Hello Fer, why do we have the validate to checking the parameters? Why not directly using them??
And also we can not implement the solution all in wollok??

**Comment by fdodino** [comment]
>Hello Fer, why do we have the validate to checking the parameters? Why not directly using them??
Well, there was an issue reported:
https://github.com/uqbar-project/wollok/issues/670
So I implemented a check in Range and also here, casting to Double values. 
Otherwise if you try to use an Interval with Strings, it will fail after initialization.

> And also we can not implement the solution all in wollok??
Is there a random object or class?

File: org.uqbar.project.wollok.lib/src/wollok/lang/Interval.xtend [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	
+	def void validate(WollokObject value) {
+		try {
+			value.wollokToJava(Double) as BigDecimal
+		} catch (Throwable e) {
+			try {
+				// Permitimos casteo de entero a BigDecimal
+				value.wollokToJava(Integer) as BigDecimal
+			} catch (Throwable eInt) {
+				throw new IllegalArgumentException(value.toString() + " : only reals are allowed in an Interval")
```
Comment (npasserini) [inline org.uqbar.project.wollok.lib/src/wollok/lang/Interval.xtend]: Given that we accept integers, I would change the comment from "real" to "number"

**Comment by npasserini** [comment]
>> And also we can not implement the solution all in wollok??
> Is there a random object or class?

There is no random object, so that part has to be done natively. I am not sure about adding one random object, should we discuss that?

> Seemed easier to implement it in Xtend.

"Easier" is kind of not enough :P
I would not get crazy about it, it is just that we prefer to implement in Wollok everything that is "possible".
I am not sure if it is posible to "go more native"... just commenting an idea.

> Hello Fer, why do we have the validate to checking the parameters? Why not directly using them??
Given that it is a library that will be used by novice students, I think we should maximize the validations because this allows us to detect mis-usage, and provide better feedback.

**Comment by npasserini** [comment]
My only issue with this pull request is that I am not sure that this object is necessary, it just came accross in a discussion. But maybe I am being just too conservative. 

Also it is not a big deal, just that I would not like to make the language bigger without a good reason, I like the simplicity and sort of minimality of current core classes.

Do you think that it will be frequently used for teaching examples?
If so just merge it.

File: org.uqbar.project.wollok.lib/src/wollok/lang/Interval.xtend (hunk 1 above)
Comment (fdodino) [inline org.uqbar.project.wollok.lib/src/wollok/lang/Interval.xtend]: :+1:

**Comment by fdodino** [comment]
>Also it is not a big deal, just that I would not like to make the language bigger without a good reason, I like the simplicity and sort of minimality of current core classes.

Didn't know about this principle.
I suppose I don't have to worry to have a LocalTime object  :)

**Comment by fdodino** [comment]
Main requirement is to have a random object for certain exercises. Very often they appear in initial courses. Less important is the concept of an interval object. :+1: if we have it.

**Pull request merged**